src/plot_api/plot_api.js

```diff
@@ -1624,6 +1624,11 @@ function _restyle(gd, aobj, _traces) {
                     flags.docalc = true;
                 }
 
+                // some attributes declare a 'recalc' flag
+                if(valObject.recalc) {
+                    flags.docalc = true;
```

Comment:
Down the line, this is going to require `valObject.dostyle`, `.doplot`, `.doCamera` etc etc... would we be better off with something like `valObject.editType = 'docalc'` etc? Which presumably we could use like a `flaglist` (and validate as such in `plotschema_test`)?

💯 🍻  for getting this effort going!